language/tools/vm-genesis/src/lib.rs

```diff
@@ -90,14 +97,48 @@ pub fn encode_genesis_transaction(
     ))
 }
 
-fn merge_txn_effects(
-    mut effects_1: TransactionEffects,
-    effects_2: TransactionEffects,
-) -> TransactionEffects {
-    effects_1.resources.extend(effects_2.resources);
-    effects_1.modules.extend(effects_2.modules);
-    effects_1.events.extend(effects_2.events);
-    effects_1
+fn convert_changeset_and_events(
```

Comment:
This is the ugly part of this approach. I essentially had to make a copy of `txn_effects_to_writeset_and_events`.